examples/data-objects/inventory-app/src/legacySharedTreeInventoryList.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	Change,
+	ChangeNode,
+	Definition,
+	SharedTree as LegacySharedTree,
+	SharedTreeEvent,
+	StablePlace,
+	TraitLabel,
+} from "@fluid-experimental/tree";
+import { DataObject, DataObjectFactory } from "@fluidframework/aqueduct";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+
+import { IInventoryListUntyped, IPart } from "./interfaces";
+
+const legacySharedTreeKey = "legacySharedTree";
+
+/**
+ * An inventory list implemented using LegacySharedTree.
+ */
+export class LegacySharedTreeInventoryList extends DataObject implements IInventoryListUntyped {
+	private _tree: LegacySharedTree | undefined;
+	private get tree() {
+		if (this._tree === undefined) {
+			throw new Error("Not initialized properly");
+		}
+		return this._tree;
+	}
+
+	protected async initializingFirstTime() {
+		const legacySharedTree = this.runtime.createChannel(
+			undefined,
+			LegacySharedTree.getFactory().type,
+		) as LegacySharedTree;
+
+		// Initialize the inventory with two parts at zero quantity
+		// REV: Is ChangeNode appropriate here? Or should I use some other type (BuildNode?)?
```

Comment:
it's not harmful to use ChangeNode, but BuildNode is more ergonomic as it'll let you omit the identifiers. You can also omit the traits key on nodes that are empty.

Reply:
Also just realized `BuildNode.definition` is a `string` rather than a `Definition` which is nice to avoid all the casting.